src/content/index.tsx

```diff
@@ -1,9 +1,167 @@
+  //console.log('Attempting to select dropdown/button!')
+
+  let dropdown = document.querySelector(query) as HTMLInputElement;
+
+  if (dropdown) {
+    //console.log('Dropdown found, clicking to open');
+    dropdown.click(); // Open the dropdown
+  } else {
+    console.log('Dropdown/button not found');
+  }
+}
+
+//Function to select the list of query for you!
+
+//the query is the querylist you want to select
+//the index is the option you want to select from top to bottom
+function selectQueryList(querylist: string, index: number) {
+  const options = document.querySelectorAll(querylist);
+
+  if (options.length > 0) {
+    //console.log('Options found! Clicking the option ' + index);
+    (options[index] as HTMLElement).click();
+
+  } else {
+    console.log('Options not found');
+  }
+}
+
 // Define the type of App as a React component
+function startAutoFill() {
+
+  // Function to select the "Start Date within" dropdown
+  function startAutoFill() {
+    setTimeout(() => {
+      selectQueryList('div[data-automation-id="multiselectInputContainer"][width="280"][dir="ltr"]', 0); //select the dropdown "Start Date within"
+    }, 100);
+    
+    setTimeout(() => {
+      selectQueryList('div[data-automation-id="promptLeafNode"][data-automation-checked="Not Checked"][height="32px"][data-uxi-widget-type="multiselectlistitem"]', 0); //select the "Future Periods" option
+    }, 600);
+
+    setTimeout(() => {
+      selectQueryList('div[data-uxi-widget-type="multiselectlistitem"][data-automation-id="promptLeafNode"][data-automation-checked="Not Checked"][height="32px"][data-uxi-multiselectlistitem-type="2"]', 1); //select the "2024-25 UBC-V Academic Year" option
+    }, 1100);
+
+    setTimeout(() => {
+      selectQuery('div[data-automation-id="promptOption"][data-automation-label="2024-25 Winter Term 1 (UBC-V) (2024-09-03-2024-12-06)"]'); //select the "2024-25 Winter Term 1 (UBC-V) (2024-09-03-2024-12-06)" option
+    }, 1600);
+
+    setTimeout(() => {
+      selectQuery('div[data-automation-id="promptOption"][data-automation-label="2024-25 Winter Term 2 (UBC-V) (2025-01-06-2025-04-08)"]'); //select the "2024-25 Winter Term 2 (UBC-V) (2025-01-06-2025-04-08)" option
+    }, 2100);
+
+    setTimeout(() => {
+      selectQueryList('div[data-automation-id="multiselectInputContainer"][width="280"][dir="ltr"]', 1); //select the dropdown "Academic Level"
+    }, 2600);
+
+    setTimeout(() => {
+      selectQuery('div[data-automation-label="Undergraduate"]'); //select the "Undergraduate" option
+    }, 3100);
+
+    /* Other options to select
+    'div[data-automation-label="Undergraduate"]' -> Undergraduate
+    'div[data-automation-label="Graduate"]' -> Graduate
+    'data-automation-label="Academic Level Not Applicable"' -> Academic Level Not Applicable
+    */
+
+    setTimeout(() => {
+      selectQuery('button[data-automation-id="wd-CommandButton_uic_okButton"]'); //select the "OK" button
+    }, 3600);
```

Comment:
Using Promises instead of fixed timeouts can make the autoselect process faster. 

https://stackoverflow.com/questions/5525071/how-to-wait-until-an-element-exists

 ```ts
function waitForElm(selector: string) {
  return new Promise(resolve => {
    if (document.querySelector(selector)) {
      return resolve(document.querySelector(selector));
    }

    const observer = new MutationObserver(mutations => {
      if (document.querySelector(selector)) {
        observer.disconnect();
        resolve(document.querySelector(selector));
      }
    });

    observer.observe(document.body, {
      childList: true,
      subtree: true
    });
  });
}

function waitAndClick(selector: string): void {
  waitForElm(selector).then((element) => {
    (element as HTMLElement).click();
  });
}
```

then you can do 

```ts
    (dropDowns[0] as HTMLElement).click(); // open time dropdown
    waitAndClick('[data-automation-id="promptOption"][data-automation-label="Future Periods"]');
    waitAndClick('[data-automation-label="2024-25 UBC-V Academic Year"]'); // select UBC V
    waitAndClick('[data-automation-id="promptOption"][data-automation-label="2024-25 Winter Term 1 (UBC-V) (2024-09-03-2024-12-06)"]'); // select Winter Term 1
    waitAndClick('[data-automation-id="promptOption"][data-automation-label="2024-25 Winter Term 2 (UBC-V) (2025-01-06-2025-04-08)"]'); // select Winter Term 2
    (dropDowns[1] as HTMLElement).click(); // open level dropdown
    waitAndClick('[data-automation-label="Undergraduate"]'); // select Undergraduate
```